util/network-devp2p/src/host.rs

```diff
@@ -1218,51 +1298,85 @@ impl IoHandler<NetworkIoMessage> for Host {
 	}
 }
 
-fn save_key(path: &Path, key: &Secret) {
+trait DiskEntity: FromStr {
+	fn path() -> &'static str;
```

Comment:
the rationale for not being associated constans?

Reply:
Good catch

Reply:
Yes, this static strings should just be constants and have some documentation `path` and `desc` are really not helpful.
Something like:
```rust
/// An entity that can be `save`d to and `load`ed from disk.
trait DiskEntity: core::str::FromStr {
  /// Path segment added to base path to discriminate entities of different kinds.
  const PATH: &'static str;
  /// A dev-readable name of that entity to be displayed in logs.
  const NAME: &'static str;
  
  /// Serialize this entity to `String`.
  ///
  /// Deserialization is done via `FromStr` trait.
  fn to_string(&self) -> String;
}
```